### PR TITLE
Apply a filter when we are updating a translation for a Gutenberg block

### PR DESCRIPTION
**Pull request by brucepearson**
Apply a filter when we are updating a translation for a Gutenberg block - https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-6941

### CODE REVIEW

File: src/strings-in-block/class-html.php [hunk 1]
```diff
@@ -89,7 +91,9 @@ public function update( \WP_Block_Parser_Block $block, array $string_translation
 				isset( $string_translations[ $string_id ][ $lang ] ) &&
 				ICL_TM_COMPLETE == $string_translations[ $string_id ][ $lang ]['status']
 			) {
-				$block->innerHTML = $string_translations[ $string_id ][ $lang ]['value'];
+				$translation = $string_translations[ $string_id ][ $lang ]['value'];
+				$block = apply_filters( 'wpml_updating_block_translation', $block, $translation );
+				$block->innerHTML = $translation;
```
Comment (andreasciamanna) [inline src/strings-in-block/class-html.php:96]: @brucepearson I think you can extract the two lines into a private method.

Something like that:

```php
private function filter_block_translation( $block, $translation ) {
	$block            = apply_filters( 'wpml_updating_block_translation', $block, $translation );
	$block->innerHTML = $translation;

	return $block;
}
```

Then here an above you can call `$block = $this->filter_block_translation( $block, $translation );`.

(or even pass `$block` by reference and avoid to return it)

**Review by andreasciamanna — APPROVED** [review]
It looks fine to me. I've only added a comment to prevent duplication (especially because of the duplicated reference to the same filter).

**Review by strategio — APPROVED** [review]
@brucepearson, I agree with Andrea, it's better if we can avoid the duplication. Please also add a PHPDoc to this new filter hook.

**Comment by brucepearson** [comment]
Closing this as there's already a filter that can be used.

**Pull request closed without merging**